modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/view/PlayerContainerFragment.kt

```diff
@@ -208,6 +258,35 @@ class PlayerContainerFragment : BaseFragment(), HasBackstack {
             menuRes = null,
             fragmentManager = parentFragmentManager,
         )
+
+
+        viewModel.networkType.observe(viewLifecycleOwner) { type ->
+            when (type) {
+
+                ConnectivityManager.TYPE_WIFI -> {
+                    Timber.tag("Check Data Connectivity").d("Connect to wifi 😊")
+                    println("Connect to wifi ")
+
+                }
+
+                ConnectivityManager.TYPE_MOBILE -> {
+                    //Log Purposes
+                    Timber.tag("Check Data Connectivity").d("Connect to Mobile Internet 😊")
+                    println("Connect to Mobile Internet ")
+                    //Toast when user is using Mobile Internet
+                    Toast.makeText(context, "Using Mobile Data", Toast.LENGTH_SHORT).show()
+
+                }
+
+                else -> {
+                    Timber.tag("Check Data Connectivity").d("Neither Connected to Wifi nor Mobile" +
+                            " Internet")
+
+
+                }
+            }
+        }
+        viewModel.checkNetworkType(requireContext())
```

Comment:
I'm not sure that the `PlayerContainerFragment` is the right place for this functionality. It doesn't seem like whether we show this should be tied to this view. Instead, I would think we'd want to give the warning anytime playback is happening, so maybe `MainActivity` or `PocketCastsApplication` would make more sense.